src/dotnet-serve/Program.cs

```diff
@@ -33,5 +47,166 @@ public static int Main(string[] args)
                 return 2;
             }
         }
+
+        static void ReadConfig(Config config, CommandLineOptions model)
+        {
+            // Port is special since it gets a value already even if none is provided,
+            // so we use the config value as an override instead.
+            if (config.TryGetNumber("serve", "port", out var port))
+            {
+                model.Port = (int)port;
+            }
+
+            if (model.Directory == null)
+            {
+                model.Directory = config.GetString("serve", "directory");
+            }
+            if (model.OpenBrowser == null)
+            {
+                model.OpenBrowser = config.GetBoolean("serve", "open-browser");
+            }
+            if (model.Quiet == null)
+            {
+                model.Quiet = config.GetBoolean("serve", "quiet");
+            }
+            if (model.Verbose == null)
+            {
+                model.Verbose = config.GetBoolean("serve", "verbose");
+            }
```

Comment:
Seems like a good place for null coalescing.
```suggestion
            model.Directory ??= config.GetString("serve", "directory");
            model.OpenBrowser ??= config.GetBoolean("serve", "open-browser");
            model.Quiet ??= config.GetBoolean("serve", "quiet");
            model.Verbose ??= config.GetBoolean("serve", "verbose");
```

etc... you get the idea.